packages/@aws-cdk/aws-cognito/test/user-pool-client.test.ts

```diff
@@ -42,6 +42,72 @@ describe('User Pool Client', () => {
     expect(() => client2.userPoolClientName).toThrow(/available only if specified on the UserPoolClient during initialization/);
   });
 
+  describe('Client with secret', () => {
+
+    test('generate secret', () => {
+      // GIVEN
+      const stack = new Stack();
+      const pool = new UserPool(stack, 'Pool');
+
+      // WHEN
+      const clientWithSecret = new UserPoolClient(stack, 'clientWithSecret', {
+        userPool: pool,
+        generateSecret: true,
+      });
+
+      // THEN
+      expect(clientWithSecret.userPoolClientSecret).toBeDefined();
+
+      // Make sure getter returns the same secret regardless if it's called one or many times
+      expect(clientWithSecret.userPoolClientSecret).toEqual(clientWithSecret.userPoolClientSecret);
+
+      // Make sure the generated template has correct Secret Generation flag
+      Template.fromStack(stack).hasResourceProperties('AWS::Cognito::UserPoolClient', {
+        GenerateSecret: Match.exact(true),
+      });
+    });
```

Comment:
This is already tested elsewhere in this file. I was meaning the resources that should be generated in this case but not the others. Like this:

```suggestion
      // Make sure the generated template has correct resources
      Template.fromStack(stack).hasResourceProperties('Custom::DescribeCognitoUserPoolClient', {
        ServiceToken: {
          'Fn::GetAtt': [
            'AWS679f53fac002430cb0da5b7982bd22872D164C4C',
            'Arn',
          ],
        },
        Create: {
          'Fn::Join': [
            '',
            [
              '{"region":"',
              {
                Ref: 'AWS::Region',
              },
              '","service":"CognitoIdentityServiceProvider","action":"describeUserPoolClient","parameters":{"UserPoolId":"',
              {
                Ref: 'PoolD3F588B8',
              },
              '","ClientId":"',
              {
                Ref: 'clientWithSecretD25031A8',
              },
              '"},"physicalResourceId":{"id":"',
              {
                Ref: 'clientWithSecretD25031A8',
              },
              '"}}',
            ],
          ],
        },
        InstallLatestAwsSdk: true,
      });

      Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
        PolicyDocument: {
          Statement: [{
            Action: 'cognito-idp:DescribeUserPoolClient',
            Effect: 'Allow',
            Resource: {
              'Fn::GetAtt': [
                'PoolD3F588B8',
                'Arn',
              ],
            },
          }],
          Version: '2012-10-17',
        },
        PolicyName: 'clientWithSecretDescribeCognitoUserPoolClientCustomResourcePolicyCDE4AB00',
        Roles: [{ Ref: 'AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2' }],
      });

      Template.fromStack(stack).hasResourceProperties('AWS::IAM::Role', {
        AssumeRolePolicyDocument: {
          Statement: [{
            Action: 'sts:AssumeRole',
            Effect: 'Allow',
            Principal: {
              Service: 'lambda.amazonaws.com',
            },
          }],
          Version: '2012-10-17',
        },
        ManagedPolicyArns: [{
          'Fn::Join': [
            '',
            [
              'arn:',
              {
                Ref: 'AWS::Partition',
              },
              ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
            ],
          ],
        }],
      });

      Template.fromStack(stack).hasResourceProperties('AWS::Lambda::Function', {
        Code: {
          S3Bucket: {
            'Fn::Sub': 'cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}',
          },
          S3Key: '105b4f39ae68785e705640aa91919e412fcba2dd454aca53412747be8d955286.zip',
        },
        Role: {
          'Fn::GetAtt': [
            'AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2',
            'Arn',
          ],
        },
        Handler: 'index.handler',
        Runtime: 'nodejs14.x',
        Timeout: 120,
      });
    });
```